### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by gusatb**
Added files to project.

### CODE REVIEW

**Comment by gusatb** [comment]
@kirbyk @C-D-Lewis Please Review

File: appinfo.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+  "uuid": "13bcd0eb-009e-4a7d-b11a-a42c8f3983bb",
+  "shortName": "to_square_example",
```
Comment (C-D-Lewis) [inline appinfo.json:3]: Change both the `name` fields to "Dynamic PDC Example"

File: src/to_square_example.c [hunk 2]
```diff
@@ -0,0 +1,177 @@
+#include <pebble.h>
+
```
Comment (C-D-Lewis) [inline src/to_square_example.c]: Remove the two- and three- newline gaps.

File: src/to_square_example.c [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+#define ABS(a) (((a) > 0) ? (a) : -1 * (a))
+#define IMAGES_LENGTH 2
+
+
+
+static Window *s_main_window;
+static Layer *icon_layer;
+static GDrawCommandImage* images [IMAGES_LENGTH];
```
Comment (C-D-Lewis) [inline src/to_square_example.c]: The `*` should be next to the variable name, not the type name. Also remove spaces between array names and lengths. 

`static GDrawCommandImage *images[IMAGES_LENGTH];`

File: src/to_square_example.c [hunk 4]
```diff
@@ -0,0 +1,177 @@
+#include <pebble.h>
+
+
+#define ABS(a) (((a) > 0) ? (a) : -1 * (a))
+#define IMAGES_LENGTH 2
+
+
+
+static Window *s_main_window;
+static Layer *icon_layer;
```
Comment (C-D-Lewis) [inline src/to_square_example.c]: Keep up the style of `s_name` as you have done with the `Window` variable for all other `static` variables.

File: src/to_square_example.c [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+static GDrawCommandImage* images [IMAGES_LENGTH];
+
+
+static int to_square_normalized;
+static int current_image;
+
+static int16_t prv_int_attract_to(int16_t i, int16_t bounds, int32_t normalized) {
```
Comment (C-D-Lewis) [inline src/to_square_example.c]: Update the names of the functions copied from the other example to more friendly names that describe what they do in the context of this app. For example, I'd call this one `update_value_from_animation_progress()`.

Comment (C-D-Lewis) [inline src/to_square_example.c]: Also, add a couple of line comments explaining what this part of the code actually does.

File: src/to_square_example.c [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+  return point;
+}
+
+typedef struct {
+  GSize size;
+  int32_t normalized;
+} ToSquareCBContext;
+
+static bool prv_attract_draw_command_list_to_square_cb(GDrawCommand *command, uint32_t index, void *context) {
+
```
Comment (C-D-Lewis) [inline src/to_square_example.c]: Remove this spurious newline.

File: src/to_square_example.c [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+  };
+  gdraw_command_list_iterate(list, prv_attract_draw_command_list_to_square_cb, &ctx);
+}
+
+void attract_draw_command_image_to_square(GDrawCommandImage *image, int32_t normalized) {
+  attract_draw_command_list_to_square(gdraw_command_image_get_command_list(image), gdraw_command_image_get_bounds_size(image), normalized);
+}
+
+//Update callback for the background layer
+
```
Comment (C-D-Lewis) [inline src/to_square_example.c]: Remove these newlines so the comment is directly above the function. Also, add a space after the forward slashes:

`// Update callback <etc>`

File: src/to_square_example.c [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+static void icon_layer_update_proc(Layer *layer, GContext *ctx) {
+  if (!images[current_image]) {
+    return;
+  }
+
+  GDrawCommandImage *temp_copy = gdraw_command_image_clone(images[current_image]);
+  attract_draw_command_image_to_square(temp_copy, to_square_normalized);
+  graphics_context_set_antialiased(ctx, true);
+  gdraw_command_image_draw(ctx, temp_copy, GPoint(0, 0));
+  free(temp_copy);
```
Comment (C-D-Lewis) [inline src/to_square_example.c]: You should not directly free memory given to you by the API. Use `gdraw_command_image_destroy()` instead.

File: src/to_square_example.c [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+static void stopped_animation_handler(Animation *animation, bool finished, void *context);
+
+static void schedule_animation_to_square(){
+  Animation * animation_to_square = (Animation *) property_animation_create(&s_icon_square_normalized_implementation, images[0], NULL, NULL);
+  animation_set_duration(animation_to_square, 1200);
+  animation_set_curve(animation_to_square, AnimationCurveEaseInOut);
+  animation_set_handlers(animation_to_square, (AnimationHandlers) {
+    .stopped = stopped_animation_handler,
+  }, NULL);
+
```
Comment (C-D-Lewis) [inline src/to_square_example.c]: Remove all these spurious newlines.

Comment (C-D-Lewis) [inline src/to_square_example.c]: And those in other functions below.

File: src/to_square_example.c [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+  s_main_window = window_create();
+  window_set_window_handlers(s_main_window, (WindowHandlers) {
+    .load = main_window_load,
+    .unload = main_window_unload,
+  });
+  window_stack_push(s_main_window, true);
+}
+
+static void deinit() {
+  for(int j = 0; j < IMAGES_LENGTH; j++)free(images[j]);
```
Comment (C-D-Lewis) [inline src/to_square_example.c]: Again, do not directly `free()` SDK objects. Use the corresponding `_destroy()` function instead.

**Comment by C-D-Lewis** [comment]
First pass complete! Make the suggested changes, then ping me again.

**Comment by sarfata** [comment]
Please include a README file. Explain what the example does, what are the most interesting parts and also explain how to convert svg into pdc. Having links to the developer website for relevant docs would be awesome too.

**Comment by kirbyk** [comment]
In addition to Thomas' comments I'd also make sure to include gif's of what the animation looks like as well as build instructions. `pebble build`, `pebble install --emulator basalt`, etc.

File: src/to_square_example.c (hunk 5 above)
Comment (kirbyk) [inline src/to_square_example.c]: Specifically, I'd even add comments to explain what each parameter does. For example, I have no idea what `i` does here. Maybe even give it a better name.

**Comment by gusatb** [comment]
@C-D-Lewis @kirbyk Second pass!

**Comment by C-D-Lewis** [comment]
Please word-wrap `README.md` to 80 characters. You can use [wrap plus](https://github.com/ehuss/Sublime-Wrap-Plus) to do this easily per-paragraph with Command+Option+Q.

File: README.md [hunk 11]
```diff
@@ -1,2 +1,28 @@
...
+First install all the python dependencies
+
+```sh
+pip install -r requirements.txt
+```
+
+Then use the [SVG2PDC](./tools/svg2pdc.py) tool to convert an SVG to PDC. Note that this will generate the PDC in the same directory as the source file (SVG).
+
+```sh
+python tools/svg2pdc.py resources/Pebble_50x50_Generic_weather.svg
```
Comment (C-D-Lewis) [inline README.md]: You need another line below this with ``` to close this code block

**Comment by C-D-Lewis** [comment]
Second pass complete.

**Comment by C-D-Lewis** [comment]
@gusatb Please chase @kirbyk up for a second pass, and get this merged soon. In the meantime, :+1: from me.

**Comment by kirbyk** [comment]
:+1: from me as well

**Comment by C-D-Lewis** [comment]
@gusatb Merge it!

**Pull request merged**